enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,253 @@
+* maximum number of logs per index (e.g. 30G) 
+* the maximum age (e.g. 7 days).  
+
+This policy is passed to the Elasticsearch CR for the `elasticsearch-operator` to manage the rollover policy.
+
+
+### Implementation Details
+
+#### Assumptions
+
+* ClusterLogging will expose the minimal needed set of Elasticsearch rollover policy management API in order to achieve the previously described goals
+* ClusterLogging will manage rollover as Elasticsearch index management is either restricted by Elastic licensing or not available in the opensource version of OpenDistro
+* Security will be addressed by using the OpenDistro security plugin and document level security (DLS). Details TBD.
+
+#### Data Model
+Logs of a given type are co-located to the following indices:
+
+| Log Type | Read Alias | Write Alias | Initial Index |
+|---------|-----------|---------------|---------------|
+| Node (`logs.infra`)|infra, infra.journal|infra.journal-write|infra.journal-00001|
+| Container Infra (`logs.infra`)|infra, infra.container|infra.container-write|infra.container-000001|
+| Container Application (`logs.app`)|app.logs|app.container-write|app.container-000001|
+
+**Note:** Log types are further defined in [LogForwarding](./cluster-logging-log-forwarding.md).
+
+#### ClusterLogging API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+metadata:
+  name: "instance"
+spec:
+  logStore:
+    retentionPolicy: 
+      logs.app:
+        maxAge: 7d
+      logs.infra
+        maxAge: 7d
+
+```
+#### Cluster Logging Operator
+The `cluster-logging-operator` will utilized the ClusterLogging CR retention policy to spec
+the desired aforementioned Elasticsearch CR indexManagement.
+#### Elasticsearch API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  indexManagement:
+    policies:
+    - name: infra-policy
+      phases:
+        hot:
+          actions:
+            rollover:
+              max_size:  30GB         #opinionated value on Elastic recommendations
+              max_age:   3d
+        delete:
+          min_age: 7d
+    templates:
+    - name:  infra
+      template:
+        index_patterns:
+        - node.infra*
+        settings:
+          number_of_shards: 1,        #value based on es node count policy
+          number_of_replicas: 1,      #value based on redundancy policy
+        aliases:
+        - infra  
+    seedings:
+    - prefix: node.infra              #creates node.infra-00001 aliased node.infra-write     
+      policyRef: infra-policy
+```
+#### Elasticsearch Operator
+The `elasticsearch-operator` will be modified to:
+* Expose the index management API
+* Create and seed specified index templates
+* Create the initial indices as needed
+* Block data ingestion if needed until initial index is seeded
+* Deploy curation CronJob for each entry in the specified seedings to rollover using the defined policy
```

Comment:
EO will be responsible for this now?

Reply:
We probably can get away with a single curator cronjob, not sure we need one per entry

Reply:
> We probably can get away with a single curator cronjob, not sure we need one per entry

I believe we can greatly simply our curation needs.  Do we really need the actual curator python code to perform this function?  We should be able to reduce it to something like a `curl` statement which has a singular responsibility of rolling a specific index which means one per.  Doesn't seem the "load" is significant

Reply:
> Do we really need the actual curator python code to perform this function?

no

> We should be able to reduce it to something like a curl statement

or the golang elasticsearch api, or even a generic REST api